src/azure-cli-testsdk/azure/cli/testsdk/preparers.py

```diff
@@ -14,15 +14,17 @@
 
 
 # Resource Group Preparer and its shorthand decorator
+# This preparer's traffic is not recorded. As a result when tests are run in record mode, sdk calls cannot be made.
+# Rather the deterministic resource group information is returned.
 
 class ResourceGroupPreparer(AbstractPreparer, SingleValueReplacer):
     def __init__(self, name_prefix='clitest.rg',
                  parameter_name='resource_group',
                  parameter_name_for_location='resource_group_location', location='westus',
                  dev_setting_name='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_NAME',
                  dev_setting_location='AZURE_CLI_TEST_DEV_RESOURCE_GROUP_LOCATION',
-                 random_name_length=75, key='rg'):
-        super(ResourceGroupPreparer, self).__init__(name_prefix, random_name_length)
+                 random_name_length=75, key='rg', disable_recording=True):
+        super(ResourceGroupPreparer, self).__init__(name_prefix, random_name_length, disable_recording)
```

Comment:
The intent with preparers is that none of the traffic for their creation is recorded. They are "assumed to exist" on playback. Thus, this fix should really be implemented in some kind of base class, because it should equally apply to StorageAccountPreparers and all the others.

Reply:
Sure, this makes sense. The other preparers might be complicated as they sometimes rely on nondeterministic info (at least StorageAccountPreparers), but it is definitely doable.

I will make the changes.